swupd/manifest.go

```diff
@@ -826,23 +821,31 @@ func writeIndexManifest(c *config, ui *UpdateInfo, bundles []*Manifest) (*Manife
 	}
 
 	bundleDir := filepath.Join(c.imageBase, fmt.Sprint(ui.version))
-	// add files from the chroot created in constructIndex
-	err := idxMan.addFilesFromChroot(filepath.Join(bundleDir, IndexBundle), "")
-	if err != nil {
-		return nil, err
-	}
 
-	// if the allbundles directory was created add all those bundle files
-	// to the index as well
-	metaRoot := filepath.Join(bundleDir, "full", indexAllBundleDir)
-	if _, err = os.Stat(metaRoot); err == nil {
-		err = idxMan.addFilesFromChroot(metaRoot, filepath.Join(bundleDir, "full"))
-		if err != nil {
+	// When there is no bundle chroot for the index manifest, create a tracking file and add
+	// the allbundles directory to the index manifest. The index manifest will be deleted when
+	// a bundle chroot directory exists.
+	if _, err := os.Stat(filepath.Join(bundleDir, IndexBundle)); os.IsNotExist(err) {
```

Comment:
When a bundle chroot for the index manifest exists (even if there is content in the bundle chroot), the index manifest will be deleted. Currently, index manifests do not support adding files with bundle chroots, so no functionality is lost. I do not want to enable bundle chroots for the index manifest since it is deprecated in future formats.